lib/puppet/environments.rb

```diff
@@ -246,10 +246,8 @@ def create_environment(name)
 
       configured_path = File.join(@environment_dir, name.to_s)
       env.configured_path = configured_path
-      if Puppet.settings[:report_configured_environmentpath]
+      if Puppet.settings[:versioned_environment_dirs]
```

Comment:
Sorry coming back to this. In puppet8, we have https://github.com/puppetlabs/puppet/blob/5503fe2d8e0b19efeccfe5174c08af5ddc3eb76c/lib/puppet/defaults.rb#L430-L432

So currently we always call `validated_directory` to resolve any symlinks in the environment directory https://github.com/puppetlabs/puppet/blob/7d8b294a8d2d40ef0e016c8fd9422b66059750c2/lib/puppet/environments.rb#L255-L259

However, we also have https://github.com/puppetlabs/puppet/blob/5503fe2d8e0b19efeccfe5174c08af5ddc3eb76c/lib/puppet/defaults.rb#L264-L265

So this PR would change the default behavior for `resolved_path`, which I think affects the `file` that the `resource` is associated with during reporting?

https://github.com/puppetlabs/puppet/blob/7d8b294a8d2d40ef0e016c8fd9422b66059750c2/lib/puppet/resource.rb#L337

A separate wrinkle is we have code that checks whether the `configured_path` and `resolved_path` are both set to avoid comparing the two strings:

https://github.com/puppetlabs/puppet/blob/7d8b294a8d2d40ef0e016c8fd9422b66059750c2/lib/puppet/node/environment.rb#L195-L196

So a side effect of setting `env.resolved_path = ...`, is that `paths_set` will always be true. Not sure we want to do that or if we should short-circuit the `externalize_path` method in cases where versioned code deploys is disabled?